tests/actors/rust/wasi-keyvalue-component/src/lib.rs

```diff
@@ -0,0 +1,119 @@
+wit_bindgen::generate!({
+    world: "test-keyvalue",
+    exports: {
+        "wasmcloud:bus/guest": WasiKeyvalueComponent
+    }
+});
+
+use std::str::FromStr;
+
+use crate::exports::wasmcloud::bus::guest::Guest;
+use crate::wasi::{
+    io::streams::write,
+    keyvalue::{
+        atomic::increment,
+        readwrite::{get, set},
+        types::{
+            incoming_value_consume_sync, new_outgoing_value, open_bucket,
+            outgoing_value_write_body, IncomingValue, OutgoingValue,
+        },
+    },
+};
+
+const DEFAULT_BUCKET: &str = ""; // NOTE: custom buckets are not yet supported by many providers
+const DEFAULT_KEY: &str = "test";
+const DEFAULT_TEXT_VALUE: &str = "hello world";
+const DEFAULT_NUMERIC_VALUE: u32 = 42;
+
+struct WasiKeyvalueComponent;
+
+/// Retrieve and parse bytes returned by consuming an incoming-value as produced by wasi:keyvalue
+fn read_wasi_kv_incoming_value<T: FromStr>(iv: IncomingValue) -> Result<T, String>
+where
+    <T as FromStr>::Err: std::fmt::Display,
+{
+    // Read bytes from incoming value
+    let bytes = incoming_value_consume_sync(iv)
+        .map_err(|count| format!("failed to parse incoming bytes, read [{count}]"))?;
+
+    // Convert the string-ified bytes to the T
+    let value = T::from_str(
+        String::from_utf8(bytes)
+            .map_err(|e| format!("failed to parse string from returned bytes: {e}"))?
+            .trim(),
+    )
+    .map_err(|e| format!("failed to convert string to value: {e}"))?;
+
+    Ok(value)
+}
+
+/// Create an outgoing-value to be used by wasi:keyvalue
+fn create_wasi_kv_outgoing_value<T: ToString>(t: T) -> Result<OutgoingValue, String> {
+    let outgoing_value = new_outgoing_value();
+    let stream = outgoing_value_write_body(outgoing_value)
+        .map_err(|_| format!("failed to write outgoing value"))?;
+
+    // Write out the new value
+    write(stream, t.to_string().as_bytes())
```

Comment:
This should be using functionality from https://github.com/wasmCloud/wasmCloud/tree/main/crates/actor just like all the tests already do